src/harness/reference_models/iap/iap.py

```diff
@@ -0,0 +1,654 @@
+
+  Args:
+    protected_entity: List of protection points inside protection area
+    grant_objects: a list of grant requests, each one being a dictionary 
+                   containing grant information
+    iap_margin_ppa: Pre IAP threshold value for PPA incumbent type(mW/RBW)
+    pal_records: PAL records associated with a PPA protection area 
+  """
+
+  logging.debug('$$$$ Getting GRID points for PPA Protection Area $$$$')
+
+  # Get Fine Grid Points for a PPA protection area
+  protection_points = utils.GridPolygon(protected_entity)
+
+  # Get the region type of the PPA protection area
+  ppa_region = protected_entity['ppaInfo']['ppaRegionType']
+
+  # Find the PAL records for the PAL_IDs defined in PPA records
+  ppa_pal_ids = protected_entity['ppaInfo']['palId']
+
+  for pal_record in pal_records:
+
+    if pal_record['palId'] == ppa_pal_ids[0]:
+
+      # Get the frequencies from the PAL records
+      ppa_freq_range = pal_record['channelAssignment']['primaryAssignment']
+      ppa_low_freq = ppa_freq_range['lowFrequency']
+      ppa_high_freq = ppa_freq_range['highFrequency']
+
+      # Get channels over which area incumbent needs partial/full protection
+      protection_channels = interf.getProtectedChannels(ppa_low_freq, ppa_high_freq)
+
+      # Apply IAP for each protection constraint with a pool of parallel 
+      # processes.
+      pool = Pool(processes=min(interf.NUM_OF_PROCESSES, 
+                                   len(protection_points)))
+
+      logging.debug('$$$$ Calling PPA Protection $$$$')
+      for protection_point in protection_points:
+        iapPointConstraint(protection_point, protection_channels, ppa_low_freq,
+          ppa_high_freq, grant_objects, None, None, ppa_region,
+          iap_margin_ppa, interf.ProtectedEntityType.PPA_AREA)
+      break
+
+  pool.close()
+  pool.join()
+
+  return
+
+
+def performIapForFss(protected_entity, grant_objects, 
+                     iap_margin_fss_cochannel, iap_margin_fss_blocking):
+  """Compute post IAP interference margin for FSS incumbent types
+
+  Routine to determine FSS protection criteria. Based on FSS protection 
+  channel range and TT&C flag type of FSS protection is determined.
+  FSS protection is provided over co-channel and blocking pass band
+
+  Args:
+    protected_entity: List of protection points inside protection area
+    grant_objects: a list of grant requests, each one being a tuple
+                              containing grant information
+    iap_margin_fss_cochannel: Pre IAP threshold value for FSS co-channel 
+                              incumbent type(mW/RBW)
+    iap_margin_fss_blocking: Pre IAP threshold value for FSS Blocking 
+                             incumbent type(mW/RBW)
+  """
+
+  # Get FSS T&C Flag value
+  fss_ttc_flag = protected_entity['deploymentParam'][0]['ttc']
+
+  fss_weight1 = protected_entity['deploymentParam'][0]['weight1']
+  fss_weight2 = protected_entity['deploymentParam'][0]['weight2']
```

Comment:
not part of the FSS registration data defined. So should use the default value defined

Reply:
Removed